alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetAlertsAction.kt

```diff
@@ -103,39 +107,75 @@ class TransportGetAlertsAction @Inject constructor(
             .sort(sortBuilder)
             .size(tableProp.size)
             .from(tableProp.startIndex)
+        var alertIndex = AlertIndices.ALL_ALERT_INDEX_PATTERN
+        if (!getAlertsRequest.alertIndex.isNullOrEmpty()) {
+            alertIndex = getAlertsRequest.alertIndex
+            getAlerts(searchSourceBuilder, alertIndex, actionListener, user)
+        } else if (getAlertsRequest.monitorId != null) {
+            transportGetMonitorAction.execute(
+                GetMonitorRequest(
+                    getAlertsRequest.monitorId,
+                    -3L,
+                    RestRequest.Method.GET,
+                    FetchSourceContext.FETCH_SOURCE
+                ),
+                object : ActionListener<GetMonitorResponse> {
+                    override fun onResponse(getMonitorResponse: GetMonitorResponse) {
+                        if (getMonitorResponse.monitor != null) {
+                            alertIndex = getMonitorResponse.monitor!!.dataSources.alertsIndex
+                            getAlerts(searchSourceBuilder, alertIndex, actionListener, user)
+                        }
+                    }
+
+                    override fun onFailure(t: Exception) {
+                        actionListener.onFailure(t)
```

Comment:
During failure, should we rather log the message and fallback to default index name?

Reply:
+1 on logging message. Will do that. This was existing code which was moved to a method hence overlooked its correctness.

But if the user passes an incorrect index name I don't think we should fallback to default as we might give them misleading info. They might assume we are surfacing alerts from the custom index name that they passed. It's better to throw an error which might make them evaluate what went wrong. Silent failure might be counterproductive.

Reply:
I am thinking about the stale config information being treated same everywhere. But yeah we can revisit this on the whole.